com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/model/ModuleHook.java

```diff
@@ -0,0 +1,111 @@
+package com.sap.cloud.lm.sl.mta.model;
+
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+public class ModuleHook extends VersionedEntity implements VisitableElement, NamedElement {
+
+    public enum ModuleHookPhase {
+        APPLICATION_BEFORE_STOP, APPLICATION_BEFORE_START, NONE;
+
+        public static ModuleHookPhase fromString(String hookPhase) {
+            if (hookPhase.equals("application.before-start")) {
+                return ModuleHookPhase.APPLICATION_BEFORE_START;
+            }
+            if (hookPhase.equals("application.before-stop")) {
+                return ModuleHookPhase.APPLICATION_BEFORE_STOP;
+            }
+
+            throw new IllegalStateException(MessageFormat.format("Unsupported hook phase {0}", hookPhase));
+
+        }
+
+    }
+
+    public ModuleHook(int majorSchemaVersion) {
```

Comment:
Rename the class to `Hook`. We agreed that in the future there may be global hooks and that they'll have the same structure.

Reply:
It was named Hook before the push - renamed it because to be more specific to the module. You have the point though - will rename it back to Hook